src/Microsoft.AspNetCore.SignalR.Redis/RedisHubLifetimeManager.cs

```diff
@@ -245,9 +249,9 @@ public override Task OnConnectedAsync(HubConnectionContext connection)
                 previousConnectionTask = WriteAsync(connection, message);
             });
 
-            if (connection.User.Identity.IsAuthenticated)
+            if (connection.User.Identity.IsAuthenticated && _userIdProvider.GetUserId(connection) != null)
```

Comment:
Don't check for authenticated, just check for null and store the result in a local so it's not executed twice.